### PR TITLE
fix expandable search bar issue w/ the dropdown options (quick search)

### PR DESCRIPTION
**Pull request by hotzjacobb**
**Description**

This PR addresses issue #528. The issue was actually broader than described. The expandable search bar was actually unable to use the dropdown options after any user input keystrokes. This PR fixes that and it now works as intended.

**Technical**

The existing code didn't work because it tried to modify a LitElement component from outside the component. LitElement, like similar frameworks, updates the DOM based on changes to state inside components. Also, the previous implementation placed a non-documented obligation on the consumer to know that they must write this supporting code.

This implementation fixes this by changing state from within the component.

I also would like to note that I discovered another pre-existing problem while fixing this, but that goes beyond the scope of this PR. Both `this.searchTerm` and `this.searchInput.value` hold overlapping state information for text in the search bar. I tried to see if there was a simple fix by looking at LitElement's documentation, but it seemed like it would take more digging or someone with more experience with that framework. A simple solution might be converting this to React. I can file an issue if need be.

**Testing**

See if the expandable search bar correctly takes new input after from the drop-down menu/the expandable search bar works in general.

**Evidence**

https://user-images.githubusercontent.com/34525444/115131511-85ed1d80-9fad-11eb-8a2e-593d73874ad0.mov

### CODE REVIEW

File: packages/expandable-search-bar/src/expandable-search-bar.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export default class ExpandableSearchBar extends LitElement {
 
   @property({ type: Boolean }) showsDisclosure = false;
 
+  // TODO: searchTerm should be single source of state;
+  //       searchInput.value should not also affect state
```
[the remark this answers is not in the document]
Comment (hotzjacobb) [reply packages/expandable-search-bar/src/expandable-search-bar.ts]: Thanks; used in 247c417

File: packages/expandable-search-bar/src/expandable-search-bar.ts [hunk 2]
```diff
@@ -80,6 +80,19 @@ export default class ExpandableSearchBar extends LitElement {
...
+   * Workaround for lit-element 2.2.1 not reflecting this.searchTerm 
+   * in the HTML element. updated is called when this.searchTerm's value
+   * is changed.
+   *
+   * @memberof ExpandableSearchBar
+   */
+  updated() {
+      if (this.searchInput) {
+        this.searchInput.value = this.searchTerm;
+      }
```
Comment (jbuckner) [inline packages/expandable-search-bar/src/expandable-search-bar.ts:93]: The `updated()` method gets called when any of the properties get updated so we'd want to limit this to just when `searchTerm` is updated. You can add the [`(props: PropertyValues)`](https://lit.dev/docs/components/lifecycle/#updated) argument to the method which contains all of the properties that were updated, ie `if (props.has('searchTerm')) { ... }`.

Also, could you reindent to 2 spaces?

File: packages/expandable-search-bar/src/expandable-search-bar.ts [hunk 3]
```diff
@@ -80,6 +80,19 @@ export default class ExpandableSearchBar extends LitElement {
     `;
   }
 
+  /**
+   * Workaround for lit-element 2.2.1 not reflecting this.searchTerm
```
Comment (jbuckner) [inline packages/expandable-search-bar/src/expandable-search-bar.ts:84]: This isn't really a workaround so much as how form inputs work. The `searchTerm` property isn't bound to the input's value as a reactive property so you have to do set it when the value gets updated.

**Review by jbuckner — CHANGES_REQUESTED** [review]

File: packages/expandable-search-bar/src/expandable-search-bar.ts [hunk 4]
```diff
@@ -88,11 +100,6 @@ export default class ExpandableSearchBar extends LitElement {
    */
   private clearSearch(): void {
     this.searchTerm = '';
-    /* istanbul ignore else */
-    if (this.searchInput) {
-      this.searchInput.value = '';
-      this.searchInput.focus();
```
Comment (jbuckner) [inline packages/expandable-search-bar/src/expandable-search-bar.ts:94]: We want to still keep the `searchInput` in focus after clearing the search.

File: packages/expandable-search-bar/src/expandable-search-bar.ts [hunk 5]
```diff
@@ -80,6 +81,17 @@ export default class ExpandableSearchBar extends LitElement {
     `;
   }
 
+  /**
+   * Update the DOM element after the value of `this.searchTerm` changes
+   *
+   * @memberof ExpandableSearchBar
+   */
+  updated(props: PropertyValues) {
```
Comment (jbuckner) [inline packages/expandable-search-bar/src/expandable-search-bar.ts:89]: Could you add a `: void` return type to this?

**Review by jbuckner — CHANGES_REQUESTED** [review]
I tested the branch and the functionality looks good except for focusing the search box after clearing. Update that and add the return type to `updated()` and this PR should be good to go.

**Review by jbuckner — APPROVED** [review]
Looks good! Thank you for the fix!

**Pull request merged**